pkg/notifier/beans/beans.go

```diff
@@ -0,0 +1,394 @@
+		for _, item := range notificationSettingsRequest.EnvId {
+			tempRequest = append(tempRequest, &LocalRequest{EnvId: item})
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) == 0 && len(notificationSettingsRequest.AppId) == 0 {
+		for _, item := range notificationSettingsRequest.TeamId {
+			tempRequest = append(tempRequest, &LocalRequest{TeamId: item})
+		}
+	} else if len(notificationSettingsRequest.TeamId) == 0 && len(notificationSettingsRequest.EnvId) > 0 && len(notificationSettingsRequest.AppId) > 0 {
+		for _, itemE := range notificationSettingsRequest.EnvId {
+			for _, itemA := range notificationSettingsRequest.AppId {
+				tempRequest = append(tempRequest, &LocalRequest{EnvId: itemE, AppId: itemA})
+			}
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) > 0 && len(notificationSettingsRequest.AppId) == 0 {
+		for _, itemT := range notificationSettingsRequest.TeamId {
+			for _, itemE := range notificationSettingsRequest.EnvId {
+				tempRequest = append(tempRequest, &LocalRequest{TeamId: itemT, EnvId: itemE})
+			}
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) == 0 && len(notificationSettingsRequest.AppId) > 0 {
+		for _, itemT := range notificationSettingsRequest.TeamId {
+			for _, itemA := range notificationSettingsRequest.AppId {
+				tempRequest = append(tempRequest, &LocalRequest{TeamId: itemT, AppId: itemA})
+			}
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) > 0 && len(notificationSettingsRequest.AppId) > 0 {
+		for _, itemT := range notificationSettingsRequest.TeamId {
+			for _, itemE := range notificationSettingsRequest.EnvId {
+				for _, itemA := range notificationSettingsRequest.AppId {
+					tempRequest = append(tempRequest, &LocalRequest{TeamId: itemT, EnvId: itemE, AppId: itemA})
+				}
+			}
+		}
+	} else {
+		tempRequest = append(tempRequest, &LocalRequest{PipelineId: notificationSettingsRequest.PipelineId})
+	}
+
+	return tempRequest
+}
+
+// GetIdsByTypeIndex
+// if new criteria fields are added, add a case to this function
+func (notificationConfigRequest *NotificationConfigRequest) GetIdsByTypeIndex(index selectorIndex) []*int {
+	switch index {
+	case teams:
+		return notificationConfigRequest.TeamId
+	case apps:
+		return notificationConfigRequest.AppId
+	case envs:
+		return notificationConfigRequest.EnvId
+	case clusters:
+		return notificationConfigRequest.ClusterId
+	}
+	return nil
+}
+
+// if new criteria fields are added, create a index for it here
+type selectorIndex int
+
+const teams, apps, envs, clusters selectorIndex = 0, 1, 2, 3
+
+// GenerateSettingCombinations
+// if new criteria is added , add a similar condition for that criteria index below
+func (notificationConfigRequest *NotificationConfigRequest) GenerateSettingCombinations() []*LocalRequest {
```

Comment:
this func seems like part of a util file

Reply:
same here